x-pack/plugins/uptime/public/components/synthetics/check_steps/steps_list.tsx

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiBasicTable, EuiPanel, EuiTitle } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useCallback, useState, useEffect, MouseEvent } from 'react';
+import styled from 'styled-components';
+import { useDispatch } from 'react-redux';
+import { Ping } from '../../../../common/runtime_types';
+import { pruneJourneyState } from '../../../state/actions/journey';
+import { clearPings } from '../../../state/actions';
+import { STATUS_LABEL } from '../../monitor/ping_list/translations';
+import { STEP_NAME_LABEL } from '../translations';
+import { ExpandRowColumn, toggleExpand } from './expand_row_col';
+import { StatusBadge } from '../status_badge';
+import { StepDetailLink } from '../../common/step_detail_link';
+import { VIEW_PERFORMANCE } from '../../monitor/synthetics/translations';
+import { StepImage } from './step_image';
+
+export const SpanWithMargin = styled.span`
+  margin-right: 16px;
+`;
+interface Props {
+  data: Ping[];
+  error?: Error;
+  loading: boolean;
+}
+
+function isStepEnd(step: Ping) {
+  return step.synthetics?.type === 'step/end';
+}
+interface StepStatusCount {
```

Comment:
nit: can we put the interface above the function so both interfaces are together?